CRX Whats New updates for March (#2557)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/whatsnew/index.md

Check this page often to learn about changes to the Chrome extensions platform, its documentation,
and related policy or other changes.

> ### Chrome 102: New manifest field "optional_host_permissions" {: #m102-optional-host-permissions }

Reviewer: Correct.  This lists them as they land in Canary.

> ### Chrome 102: New manifest field "optional_host_permissions" {: #m102-optional-host-permissions }

Author: For some additional context, we include Canary changes in order to encourage developers to test provide feedback as early as possible. If we did not call out items here, developers would have to monitor Chromium commits themselves to find out extensions platform changes before they landed in Stable.

April 4, 2022

> Added a new manifest key named `optional_host_permissions` for Manifest V3. This behaves the exact

Author: Rephrased this item using your suggested intro and tried to clarify the second sentence. The updated copy reads

> Manifest V3 extensions can now specify the `optional_host_permissions` key in manifest.json. This
allows Manifest V3 extensions to declare optional match patterns for hosts just as Manifest V2
extensions could using the `optional_permissions` key.

> Added a new manifest key named `optional_host_permissions` for Manifest V3. This behaves the exact

Author: We've been using "Manifest V3". I'm very surprised to hear that lower case outnumbered upper case. I just did a case sensitive search on my local clone and saw very different results. 

| Search string | results | files | 
| --- | --- | --- |
| Manifest v3 | 2 | 1 |
| manifest v3 | 3 | 1 |
| Manifest V3 | 197 | 31 |
| manifest V3 | 0 | 0 |

I just created PR #2623 to address these inconsistencies.

same as specifying hosts in `optional_permissions` in Manifest V2.

### Chrome 102: injectImmediately property in scripting.executeScript() {: #m102-injectimmediately }

April 4, 2022

> `chrome.scripting.executeScript()` now accepts an optional `injectImmediately` property.  If present

Reviewer: This is a property on the ScriptInjection dictionary, which is the argument - injectImmediately is not itself an argument.

https://developer.chrome.com/docs/extensions/reference/scripting/#type-ScriptInjection

> `chrome.scripting.executeScript()` now accepts an optional `injectImmediately` property.  If present

Author: PR #2624 contains a small revision to make this more clear. Please continue discussion in that PR.

and set to true, the script will inject into the target as soon as possible, rather than waiting for
`document_idle`. Note that this is not a guarantee the script will inject before the page is loaded
since the page continues to load while the API call is being made.

### Chrome 102: Omnibox API support in Manifest V3 {: #m102-omnibox }

March 31, 2022

> The [Omnibox API](/docs/extensions/reference/omnibox) can now be used in service worker based

Reviewer: ```suggestion
The [Omnibox API](/docs/extensions/reference/omnibox) can now be used in service worker-based
```

extensions. Previously, some of this API's methods would throw on invocation due to internal
dependencies on DOM capabilities.

### Chrome 102: wasm-unsafe-eval allowed in Manifest V3 CSP {: #m102-wasm }

March 22, 2022

> Manifest V3 extensions can now include `wasm-unsafe-eval` in their `content_security_policy` declarations. This change allows Manifest V3 extensions to use Wasm.

Reviewer: nit 1: wrap at 80 char
nit 2: maybe avoid "Wasm" since it's not defined?

```suggestion
Manifest V3 extensions can now include `wasm-unsafe-eval` in their `content_security_policy`
declarations. This change allows Manifest V3 extensions to use Web Assembly.
```

Author: Ack.

Author: Slight tweak to your suggestion: remove the space between "Web" and "Assembly".

```suggestion
Manifest V3 extensions can now include `wasm-unsafe-eval` in their `content_security_policy`
declarations. This change allows Manifest V3 extensions to use WebAssembly.
```

### Docs update: Chrome Web Store item discovery {: #cws-discoevery-doc }

March 21, 2022

[Discovery on Chrome Web Store](/docs/webstore/discovery/) gives an overview of how users find items
on the Chrome Web Store and how our editors select items to feature.

### Chrome 101: Improved declarativeNetRequest domain conditions {: #m101-dnr-conditions }

March 9, 2022

[declarativeNetRequest](/docs/extensions/reference/declarativeNetRequest/) rule conditions have been
updated to allow extensions to better target requests based on the request's "request" and
"initiator" domains. The relevant condition properties are `initiatorDomains`,

> `excludedInitiatorDomains`, `requestDomains`, and `excludedRequestDomains`. Also, the engineer that
> implemented this change announced this change and answered some questions form developers in this
> [chromium-extensions
> thread](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/4971ZS9cI7E). 

Reviewer: This seems a little too verbose for a brief what's-new entry.  Maybe:

```suggestion
`excludedInitiatorDomains`, `requestDomains`, and `excludedRequestDomains`. See also this
[chromium-extensions
thread](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/4971ZS9cI7E). 
```

Author: Thanks for trimming the fat here!

### Chrome 100: Resolved issue with scripting.executeScript on newly created tabs {: #m100-executescript-bugfix }

> There was a longstanding issue where calling `scripting.executeScript()` on a newly created tab or
> window could fail. These should be resolved now and it should execute as you would expect.

Reviewer: Shorter:

```suggestion
Fixed a longstanding issue where calling `scripting.executeScript()` on a newly created tab or
window could fail.
```

Saying "we fixed it and it should now work as you expect" seems redundant (since I'd assume "fixing it" requires "making it work as I'd expect")

> window could fail. These should be resolved now and it should execute as you would expect.

Author: The affected line line was removed from the final version of the PR.

Web Store review process and explains how [developer program
policy](/docs/webstore/program_policies/) enforcement is handled.

### Chrome 98: scripting.executeScript and scripting.insertCSS accept multiple files {: #m98-execute-multiple-files }

November 9, 2021

The Scripting API's [`executeScript()`](/docs/extensions/reference/scripting/#method-executeScript)

> and [`insertCSS()`](/docs/extensions/reference/scripting/#method-insertCSS) methods now accept an
> array of files. Previously these methods accepted an array of files, but the array could only
> contain a single file string. 

Reviewer: ```suggestion
and [`insertCSS()`](/docs/extensions/reference/scripting/#method-insertCSS) methods now accept multiple
files. Previously these methods required an array with a single file entry.
```
